back/src/workout/workout.service.ts

```diff
@@ -28,14 +32,21 @@ export class WorkoutService {
     private readonly exerciseTemplateRepository: ExerciseTemplateRepository,
     @Inject(EXERCISE_REPOSITORY)
     private readonly exerciseRepository: ExerciseRepository,
+    @Inject(PROGRAM_REPOSITORY)
+    private readonly programRepository: ProgramRepository,
   ) {}
 
   async create(workoutInput: WorkoutInput): Promise<Workout> {
-    const workout = new Workout({
+    const workoutInstance = new Workout({
       id: uuid(),
       title: workoutInput.title,
     })
-    return this.workoutRepository.save(workout)
+    const workout = await this.workoutRepository.save(workoutInstance)
+    await this.programRepository.saveWorkoutToProgram(
+      workoutInput.programId,
+      workout,
+    )
+    return workout
   }
 
```

Comment:
Double check with this documentation : https://typeorm.io/many-to-one-one-to-many-relations